### ISSUE TITLE
V1 addon (TypeScript) with app/components/some-component.ts fails to find the component

### ISSUE THREAD

**Issue by raido** [embroider-build/embroider#1857]
V1 addon with structure like this:

```ts
addon/components
   some-component.ts
   some-component.hbs
app/components
  some-component.ts // which just re-exports addon/components with absolute path (package-name/addon/...);
```

Fails to build (embroider-safe, embroider-optimized scenarios) with errror:

`Attempted to resolve 'some-component', which was expected to be a component, but nothing was found.`

**Workaround**

Renaming `app/components/some-component.ts` -> `app/components/some-component.js` and the build succeeds.

**Context**

```ts
@embroider/core 2.1.1 
@embroider/webpack 2.1.1 
@embroider/compat 2.1.1
webpack 5.91.0
ember-source ~4.12.0

// ember-cli-build.js

const app = new EmberAddon(defaults, {
    "ember-cli-babel": {
      enableTypeScriptTransform: true,
    },
}
```

**Comment by NullVoxPopuli** [comment]
in your addon, do you have:
```
options: {
  'ember-cli-babel': {
    enableTypeScriptTransform: true
  }
}
```
?

**Comment by raido** [comment]
Yes, I do. The whole context up 🔝 is from the addon perspective.

**Comment by Windvis** [comment]
Isn't this one of the known caveats of the TS support in Ember? 

- https://github.com/typed-ember/ember-cli-typescript?tab=readme-ov-file#account-for-addon-build-pipeline-changes
- https://github.com/typed-ember/ember-cli-typescript/issues/780

I wouldn't expect this to work under classic either then, but maybe that info is outdated?

@raido is this only a problem under Embroider? Or does it just fail in a different way?

**Comment by ef4** [comment]
Yes, even in classic this isn't really supported. But it can happen to work if the app also has typescript. 

Whereas under embroider we keep them more apart and will never apply the app's custom transpile to the addon's app folder.
